src/dali_executor/dali_executor.cc

```diff
@@ -148,7 +154,12 @@ std::vector<OutputInfo> DaliExecutor::Run(const std::vector<IDescr>& inputs) {
     ret[out_idx] = {outputs_shapes[out_idx], pipeline_.GetOutputType(out_idx),
                     pipeline_.GetOutputDevice(out_idx)};
   }
-  inputs_consumed_ = true; // this will change with introduction of streamed input
+  for (auto &name: input_names_) {
+    auto trace = pipeline_.GetOperatorTrace(name, "next_output_data_id");
+    if (trace != std::to_string(request_id_)) {
```

Comment:
```suggestion
    if (std::stoill(trace) != request_id_) {
```
It is a bit more intuitive to compare ints than strings.

Reply:
Or it won't work if there is no trace available?

Reply:
Done